scripts/Check-OfflineFilesList.ps1

```diff
@@ -0,0 +1,41 @@
+# This script checks that the list of offline files precached by the Service Worker is complete
+
+# List below any files in www that are not required by the offline precache
+$ListOfExemptions = (
+  'www/img/icons/kiwix-120.png',
+  'www/img/icons/kiwix-128.png',
+  'www/img/icons/kiwix-16.png',
+  'www/img/icons/kiwix-19.png',
+  'www/img/icons/kiwix-38.png',
+  'www/img/icons/kiwix-48.png',
+  'www/img/icons/kiwix-64.png',
+  'www/img/icons/kiwix-90.png',
+  'www/img/Kiwix_icon_transparent_600x600.png',
+  'www/js/lib/webpHeroBundle_0.0.0-dev.27.js'
+)
+
+# Get the root directory
+$RepoRoot = $PSScriptRoot -replace '[\\/]scripts'
+
+# List all files recursively in the /www directory, and process the FullName to remove the path up to 'www'
+# and to replace any (back)slashes with forward slashes
+$ListOfFSFiles = ls -r $RepoRoot/www/*.* | % { $_.FullName -replace '^.+?[\\/](?=www)', '' -replace '[\\/]', '/' }
+# Select lines matching "www/.+" in service-worker.js and process the output to replace whitespace, quotation marks and commas 
+$ListOfSWFiles = (sls '[''"]www/.+[''"]' $RepoRoot/service-worker.js) | % { $_.Line -replace '\s*[''"],?', '' }
+# Flag to track any missing files
+$MissingFiles = $false
+# Iterate the list of files found in the FS and check if they are either listed in the Service Worker or in the list of exemptions
+# NB The operator -ccontains is case-sensitive
+$ListOfFSFiles | % {
+  if (-Not ($ListOfSWFiles -ccontains $_ -or $ListOfExemptions -contains $_)) {
+    "ERROR! The file '$_' is not in the list of offline files in service-worker.js`n"
+    $MissingFiles = $true
+  }
+}
+if ($MissingFiles) {
+  Write-Host "** Please add the missing file(s) listed above to service-worker.js **`n" -ForegroundColor red
+  exit 1
+} else {
+  Write-Host "All non-exempt files in /www are listed in service-worker.js`n" -ForegroundColor green 
```

Comment:
I disagree with the CodeFactor warning below. First, we know the context this script will mostly run in (GitHub shell), where it works. Second, `Write-Host` is the only command that allows setting colours in the terminal (the recommended as substitutes do not accept the `-ForegroundColor` switch). Finally, the output is captured perfectly well by the GitHub Web UI. The script can also be run locally by the user (in Terminal), and is confirmed to work on W10 and W11.